modules/apps/blogs/blogs-service/src/main/java/com/liferay/blogs/internal/model/listener/UserGroupModelListener.java

```diff
@@ -75,13 +79,42 @@ private void _unsubscribeUserGroupUsersFromGroup(
 			long groupId, long userGroupId)
 		throws PortalException {
 
-		for (long userId :
-				_userGroupLocalService.getUserPrimaryKeys(userGroupId)) {
+		Set<Long> userGroupUserIds = SetUtil.fromArray(
+			_userGroupLocalService.getUserPrimaryKeys(userGroupId));
 
-			if (!_groupLocalService.hasUserGroup(userId, groupId)) {
-				_blogsEntryLocalService.unsubscribe(userId, groupId);
+		userGroupUserIds.removeAll(
+			SetUtil.fromArray(_groupLocalService.getUserPrimaryKeys(groupId)));
+
+		long[] organizationIds = _groupLocalService.getOrganizationPrimaryKeys(
+			groupId);
+
+		Group group = _groupLocalService.getGroup(groupId);
+
+		if (group.getOrganizationId() > 0) {
+			organizationIds = ArrayUtil.append(
+				organizationIds, group.getOrganizationId());
+		}
+
+		for (long organizationId : organizationIds) {
+			userGroupUserIds.removeAll(
+				SetUtil.fromArray(
+					_userLocalService.getOrganizationUserIds(organizationId)));
```

Comment:
Instead of creating a set just to discard it, why not use `Arrays.forEach(_userLocalService.getOrganizationIds(...), userGroupUserIds::remove)`. This will avoid creating intermediate data structures.